keras_cv/losses/giou_loss.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import tensorflow as tf
+
+from keras_cv import bounding_box
+
+
+class GIoULoss(tf.keras.losses.Loss):
+    """Implements the GIoU Loss
+
+    GIoU loss is a modified IoU loss commonly used for object detection. This loss aims
+    to directly optimize the IoU score between true boxes and predicted boxes. GIoU loss
+    adds a penalty term to the IoU loss that takes in account the area of the
+    smallest box enclosing both the boxes being considered for the iou. The length of
+    the last dimension should be atleast 4 to represent the bounding boxes. While
+    this dimension can have more than 4 values, these values will be ignored for the
+    calculation of this loss.
+
+    Args:
+        bounding_box_format: a case-insensitive string which is one of `"xyxy"`,
```

Comment:
lets not provide an exhaustive list and instead redirect to the docs for this.  That way when we update it the docstrings get updated too.